core/ir/aarch64/encode.c

```diff
@@ -198,6 +199,80 @@ const reg_id_t dr_reg_fixer[] = { REG_NULL,
 };
 /* clang-format on */
 
+/* Maps containing-registers to their DR_ISA_REGDEPS virtual-register.
+ * Same size as dr_reg_fixer[], keep them synched.
+ * Note: DR_REG_INVALID (== DR_REG_V1 == 1) has been excluded to avoid failures on opnd_t
+ * operations.
+ */
+const reg_id_t dr_reg_virtual[] = {
+    DR_REG_V0,   DR_REG_V0,   DR_REG_V2,   DR_REG_V3,   DR_REG_V4,   DR_REG_V5,
+    DR_REG_V6,   DR_REG_V7,   DR_REG_V8,   DR_REG_V9,   DR_REG_V10,  DR_REG_V11,
+    DR_REG_V12,  DR_REG_V13,  DR_REG_V14,  DR_REG_V15,  DR_REG_V16,  DR_REG_V17,
+    DR_REG_V18,  DR_REG_V19,  DR_REG_V20,  DR_REG_V21,  DR_REG_V22,  DR_REG_V23,
+    DR_REG_V24,  DR_REG_V25,  DR_REG_V26,  DR_REG_V27,  DR_REG_V28,  DR_REG_V29,
+    DR_REG_V30,  DR_REG_V31,  DR_REG_V32,  DR_REG_V33,  DR_REG_V34,  DR_REG_V2,
```

Comment:
I'm not sure I follow.
This is not a sequence, it's a mapping, even if we have DR_REG_V0 starting after DR_REG_NULL and DR_REG_INVALID, we'd have a jump there.
https://github.com/DynamoRIO/dynamorio/pull/6783#discussion_r1571489157

Reply:
I think fundamentally these arrays are hard to read by a reviewer or a future code maintainer, as evidenced by all my review comments and perhaps confusion.  The existing dr_reg_fixer are much easier to read as they have clear patterns: NULL, NULL, 0-32, etc.  I think these arrays need to have comments added identifying the original DR_REG_ value being mapped to each slot (or for each set of similar slots like GPRs etc.) as otherwise these will be too hard to read and maintain.

Reply:
Losing track of which were resolved: going to unresolve this one.

Reply:
Ok, I added the original DR_REG_ being mapped as a comment, and followed the same "define pattern" of dr_reg_fixer[] in aarch64.
I can also add "define patterns" to the other arches (note that dr_reg_fixer[] does not though, so is gonna be a bit "asymmetric") if they are still unreadable.